eng/pipelines/libraries/enterprise/linux.yml

```diff
@@ -50,7 +50,7 @@ steps:
   displayName: Test linuxclient connection to web server
 
 - bash: |
-    docker exec linuxclient bash -c '/repo/src/coreclr/build.sh -release -skipnuget -clang9 && /repo/libraries.sh /p:CoreCLRConfiguration=Release'
+    docker exec linuxclient bash -c '/repo/build.sh -subsetcategory coreclr-libraries -runtimeconfiguration release -ci'
```

Comment:
Could you add -subset to match the same behavior we had before? So that it doesn't build test tools nor packages?

it would be `-subset runtime-corelib-nativecorelib-all`

Reply:
enumerating all subsets when you want to exclude just a single one doesn't seem like a maintainable approach. maybe we need something like -nosubset. Just some thinking.

Reply:
Also how much does that actually save us in time? Is it worth the maintainability burder?

Reply:
> maybe we need something like -nosubset

Yeah that would be a good addition.

I don't know how much it saves us on time, but I think it is worth, just building what we need for these scenarios, keeps the build smaller.

Also, do we have to keep using the `-clang9` parameter? I believe coreclr by default tries to use clang5 if installed. I even see we use `-clang9` in coreclr builds for linux.

Reply:
@janvorli doesn't coreclr by default use clang9 now?

> I don't know how much it saves us on time, but I think it is worth, just building what we need for these scenarios, keeps the build smaller.

I disagree, as I believe we shouldn't maintain a list of subsets anywhere than in Subsets.props.

Reply:
Then why is the point of having the Subset feature?

Reply:
As you noticed, the current subset feature isn't mature enough for our needs. It was a direct port from core-setup with the addition of subset-categories. We should revise the subset logic and discuss alternatives, improvements. There is no point in having an x-y graph here, subsetcategories vs subsets doesn't make much sense if you can already have a subset called librariesall. Those are things we need to discuss to ease the developer workflow and the maintainability burden *in code*.

Reply:
I understand that, but these builds use hosted pools which are limited on space. Nuget packages use space, so sometimes is better to not build them when they're not needed. 

For example, the https stress build definition got broken when you enabled the src/ref restore project because it started building more things and ran out of space:
https://dev.azure.com/dnceng/public/_build/results?buildId=553931&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=abae1f68-3c73-5bff-491f-f2b908580ce6

So I think we should be careful when changing what builds do, just because the command gets more complex (while using features we recommend our developers to use). Whenever we fix the way that works, well then we go and update this build definitions here. I don't like saying, this feature is incomplete or the design is bad and then have fallouts to avoid using it because we don't like the feature we have.

Reply:
Actually that build the error was caused because the build is actually silently failing:

```
MSBUILD : error MSB1001: Unknown switch.
Switch: -skiptests

For switch syntax, type "MSBuild -help"
[91mBuild failed (exit code '1').
[0mFailed to restore the optimization data package.
The command '/bin/sh -c ./src/coreclr/build.sh -release -skiptests -clang9 &&     ./libraries.sh -c $CONFIGURATION -runtimeconfiguration release' returned a non-zero code: 1
```

and changing the build arguments made the build do work, but anyway, disk space and disk operations are expensive in builds, so that is one of the reasons to try to keep builds as minimal as possible if we don't need to build parts of the product.

Reply:
I still don't buy that. If we don't have enough disk space to produce a nuget package then we better not run the build at all as it requires to download hundreds of nuget packages and probably half a gig of disk space.

Also how did you come up with the list of subsets `runtime-corelib-nativecorelib` vs all the subsets that we run `runtime-linuxdac-corelib-nativecorelib-tools-packages`? Ie don't we need tools for crossgen to be built?

> So I think we should be careful when changing what builds do, just because the command gets more complex

It's not about the command getting more complex but about adding unnecessary maintenance burden and yet another bug farm when interacting with subsets / introducing a new subsets / removing one from the root build.

Reply:
From my understanding we don't need `linuxdac` nor `tools` nor `packages`. I've been doing `runtime-corelib-nativecorelib` and that builds crossgen and crossgens  S.P.CoreLib just fine. Anyway, I'm just trying to be consistent with what this build had initially, I don't like introducing unnecessary steps just because the way of building changed. Anyway, it seems we're not getting anywhere here. I'm just adding my two cents because I noticed a change on behavior here.